lib/posix/posix.nim

```diff
@@ -772,6 +772,12 @@ else:
   proc sigtimedwait*(a1: var Sigset, a2: var SigInfo,
                      a3: var Timespec): cint {.importc, header: "<signal.h>".}
 
+when defined(sunos) or defined(solaris):
+  # The following compile time flag is needed on Illumos/Solaris to use the POSIX
+  # `sigwait` implementation. See the documentation here:
+  # https://docs.oracle.com/cd/E19455-01/806-5257/gen-75415/index.html
```

Comment:
The "documentation here" does not mention any `POSIX_PTHREAD_SEMANTICS`.

Reply:
Sorry, I pasted the wrong link. Now it points to the right place. Besides the Oracle Solaris docs, I also added a link to the Illumos manpage of `sigwait`, containing further infos and an example.